src/config.ts

```diff
@@ -13,5 +16,13 @@ export function loadConfig(path: string | undefined) {
       }
     : { cwd: path };
 
-  return load({}, loadOptions);
+  const config = await load({}, loadOptions);
+
+  for (const rule in extendsRules) {
+    if (!config.rules?.[rule]) {
+      config.rules[rule] = extendsRules[rule];
+    }
+  }
+
+  return config;
```

Comment:
Could we pass the rules to `load` directly?

```suggestion
  return load({ rules: extendsRules || {} }, loadOptions);
```

Reply:
@joshbolduc I test this before, but it will overwrite the extended configuration setting in the workspace `commitlint.config.js`.

When the workspace `commitlint.config.js` is:
```js
module.exports = {
  extends: ['@commitlint/config-conventional'],
};
```

and the rules I was setting:
```json
"commitlint.config.extend.rules": {
  "body-leading-blank": [3, "always"],
}
```

It will load the rule of "body-leading-blank" with `[3, "always"]`. I think rules should be consistent with the workspace setting.
 And it could add rules which not include in the workspace setting. So I use the loop above to judge whether it should include. 😀

Reply:
Makes sense! In that case, I think this might read a little more clearly, and should work the same way?

```ts
const config = await load({}, loadOptions);

return { ...config, rules: { ...extendsRules, ...config.rules } };
```